docs/DeveloperGuide.md

```diff
@@ -1,34 +1,920 @@
+
+> :memo: It is set to `20` as default in actual execution. 
+
+**Step 1:** When the user enters the command for `Browse`, the input will be processed and parsed by `parser.java` and 
+then further parsed by `BrowseParser.java`.
+
+**Step 2:** Upon completion of parsing and input validation, `BrowseParser.java` will create an executable `BrowseCommand` 
+object and return it to `main`.
+
+**Step 3:** Once the `BrowseCommand` object is executed, it will firstly process the `AnimeData` by calling on its `BrowseCommand#sortBrowseList()` operation to identify the order that the list will need to be in (if any).
+
+For this case since it is a default browse operation, there is no sorting performed.
+
+**Step 4:** Now `BrowseCommand` will utilise its `BrowseCommand#buildBrowseOutput()` operation to access all 
+`Anime objects within the page window, as shown in the diagram below.
+
+![Browse Object Diagram 1](images/Browse-Default-State.png) <br/>
+*Figure 10: Browse Default State Object Diagram*
+
+In this example, it fetches the following `Anime` objects.
+```text
+Charlie
+Echo
+Gamma
+```
+
+If the 2nd page of the list was requested instead with the command `browse -p 2`. 
+`BrowseCommand#buildBrowseOutput()` will shift its page window down by 1 page as depicted in the diagram below.
+
+![Browse Object Diagram 2](images/Browse-Default-State2.png) <br/>
+*Figure 11: Browse Next Page Object Diagram*
+
+**Step 5:** At each `Anime` object, it will access its methods to get the relevant information about that anime series and construct a printable result for the user to view.
+
+**Step 6:** After all `Anime` objects in the page window have been retrieved, it will return the printable result back to `BrowseCommand#execute()`. At this point it will utilise `BrowseCommand#sortBrowseOption()` again to reset the `AnimeData` list to its original form if it has been altered.
+
+**Step 7:** Once `AnimeData` has been sorted, it will return the result back to `Main` for printing.
+
+Currently, this result is not very exciting as it is just based on the Anime ID which is assigned randomly by our data source, AniList.
+
+An example scenario would be browsing the 2nd page of a **sorted** list in ascending order.
+The only step that would change would be at Step 3, where it will perform sorting of `AnimeData` list. 
+
+![Browse Object Diagram 3](images/Browse-Sorted-State.png) <br/>
+*Figure 12: Browse Sorted State Object Diagram*
+
+As you can see, even though the page window is at the same position as the previous command, 
+the list is different as it has been sorted.
+
+From this point onwards, the operation will continue as per the steps above but during the final `BrowseCommand#sortBrowseList()` call in step 6. It will perform a sort to reset the list.
+
+Here is the sequence diagram to better illustrate the lifecycle of a browse command.
+
+![Browse Sequence Diagram](images/Browse-SequenceDiagram.png) <br/>
```

Comment:
![image](https://user-images.githubusercontent.com/57039314/97526171-884dc200-19e3-11eb-8730-56cc7f303af5.png)

Nice sequence diagram to illustrate the entire lifecycle of the browse command! I'm not sure if it's my device but there seems to be some missing arrows for return values? Or is that the intended course of action for the various methods that were called? Also, the activation bar should end at appropriate life cycles, however there are a few activation bars that are left elongated.